pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/jstl/JstlTypeConverter.java

```diff
@@ -450,6 +461,20 @@ protected OffsetDateTime coerceToOffsetDateTime(Object value) {
         MessageFactory.get("error.convert", value, value.getClass(), OffsetDateTime.class));
   }
 
+  protected BigDecimal coerceToBigDecimal(Object value) {
```

Comment:
Do we need to have converters for numeric types ?
Also should other numeric types be able to coerce from BigDecimal ?

Reply:
I could only think of String because the number + scale are self contained and gives user flexibility depending on how the value is generate in their upstream.

If it makes sense, I could coerce byte[] as well assuming it is meant to represent unscaled decimal (i.e. scale 0). I can't think of other meaningful conversions.

Reply:
It's easy to create a BigDecimal from a number:
BigDecimal.valueOf(Long)
new BigDecimal(Double)

For the conversion from bytes, I think it's not clear that the bytes are representing a BigInteger (and not a Long or a Double). So I think we shouldn't support this conversion..

Reply:
Ok, will do